test/tools/router_check/source/router.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <fstream>
+
+#include "common/common/logger.h"
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/json/json_loader.h"
+#include "common/router/config_impl.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/precompiled/precompiled_test.h"
+#include "test/test_common/utility.h"
+#include "test/tools/router_check/json/tool_config_schemas.h"
+
+/** Class that store the configuration parameters of the router
+ * check tool extracted from a json input file
+ */
+class ToolConfig {
+public:
+  ToolConfig(){};
```

Comment:
`  ToolConfig() : random_value_(0) {}` will set `random_value_` to 0 (or you could just write `int random_value_ = 0;` below. Otherwise you have a `ToolConfig` with an undefined `random_value_`.